File: Hue_Light_Project/app/src/main/java/com/minds/great/hueLightProject/utils/dagger/HueModule.java
package com.minds.great.hueLightProject.utils.dagger;

import android.content.Context;
import android.support.annotation.NonNull;

import com.minds.great.hueLightProject.core.domain.domainInterfaces.LightSystemInterface;
import com.minds.great.hueLightProject.data.HueDomain;
import com.minds.great.hueLightProject.hueImpl.HueLightSystem;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class HueModule {

    private Context context;

    public HueModule(@NonNull Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    LightSystemInterface providesLightSystemInterface(){
        return new HueLightSystem(context);
    }

    @Provides
    @Singleton
    HueDomain providesHueDomain(){
        return new HueDomain();
    }
}

File: Hue_Light_Project/app/src/main/java/com/minds/great/hueLightProject/hueImpl/HueModule.kt
package com.minds.great.hueLightProject.hueImpl

import com.minds.great.hueLightProject.core.domain.domainInterfaces.LightSystemInterface
import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@Module
@InstallIn(ActivityComponent::class)
abstract class HueModule {
    @Binds
    abstract fun bindLightSystem(hueLightSystem: HueLightSystem): LightSystemInterface
}